src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -4911,7 +4911,8 @@ protected override object PropertyGet(PSProperty property)
         /// <param name="convertIfPossible">instructs the adapter to convert before setting, if the adapter supports conversion</param>
         protected override void PropertySet(PSProperty property, object setValue, bool convertIfPossible)
         {
-            string valueString = setValue as string;
+            // XML is always a string so implicitly convert to string
+            string valueString = setValue?.ToString();
```

Comment:
Calling ToString directly avoids any magic PowerShell might apply when converting a value to string.

You could just call `LanguagePrimitives.ConvertTo<string>(setValue)` - this will properly call into a `CodeMethod` or `ScriptMethod` named `ToString` if the object has one.